pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -167,6 +279,12 @@ func (c *Controller) handleDeletePod(key string) error {
 			return err
 		}
 	}
+
+	if obj != nil {
+		newPod := obj.(*corev1.Pod).DeepCopy()
+		removePodAnnotation(newPod)
+		return c.updatePodAnnotation(newPod)
```

Comment:
At this time, the Pod is already removed from K8s API, I think it will only see NotFound error?

Reply:
so in `syncPod` I added an extra conditional for a  podkey which checks if it satisfies the `isNPLEnabledForServiceOfPod` condition.
The podKeys that do not satisfy that condition (meaning that the parent service no longer has the correct annotation), they would land up in handleRemovePod too, in which case the expectation is that the pod annotation should be removed. 
So essentially `obj == nil` means it's deleted Pod case, and `obj != nil` means that the service does not have the annotation (anymore).